netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -406,11 +430,40 @@ protected ClientTransportFactory buildTransportFactory() {
       }
       negotiator = createProtocolNegotiatorByType(negotiationType, localSslContext);
     }
+
+    boolean usingSharedGroup = eventLoopGroup == null;
+
+    populateDefaultChannelAndGroup();
+
     return new NettyTransportFactory(
         negotiator, channelFactory, channelOptions,
         eventLoopGroup, flowControlWindow, maxInboundMessageSize(),
         maxHeaderListSize, keepAliveTimeNanos, keepAliveTimeoutNanos, keepAliveWithoutCalls,
-        transportTracerFactory.create(), localSocketPicker);
+        transportTracerFactory.create(), localSocketPicker, usingSharedGroup);
+  }
+
+  private void populateDefaultChannelAndGroup() {
+    boolean groupOrChannelProvided = channelFactory != DEFAULT_CHANNEL_FACTORY
+        || eventLoopGroup != null;
+
+    if (groupOrChannelProvided) {
+      logger.log(
+          Level.WARNING,
+          "Both EventLoopGroup and ChannelType should be provided, otherwise client may not start "
+              + "depends on the gRPC default value.");
+
+      if (eventLoopGroup == null) {
+        eventLoopGroup = SharedResourceHolder.get(Utils.NIO_WORKER_EVENT_LOOP_GROUP);
+        logger.log(Level.FINE, String.format("Using fall back EventLoopGroup %s", eventLoopGroup));
```

Comment:
This will always call String.format(), even if not logging at FINE level. That should be avoided. Ditto next log.